src/Servers/IIS/IIS/src/Core/IISHttpContext.IHttpConnectionFeature.cs

```diff
@@ -81,15 +81,33 @@ string IHttpConnectionFeature.ConnectionId
         private void InitializeLocalEndpoint()
         {
             var localEndPoint = GetLocalEndPoint();
-            LocalIpAddress = localEndPoint.GetIPAddress();
-            LocalPort = localEndPoint.GetPort();
+            if (localEndPoint == null)
+            {
+                // ApplicationInitialization request might not set the local endpoint
+                LocalIpAddress = IPAddress.None;
```

Comment:
Remind me: The potential issue here is the nullref when dereferencing `localEndPoint` right? If so, maybe just don't set `LocalIpAddress` and `LocalPort` if `localEndPoint` is `null` rather than setting them to bogus values?

Reply:
I'm reviewing code now and I think you are right. If the calling code in our middleware (for example, the forwarding header middleware: https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs#L219) handles null correctly, then I think we are okay to return null.